private-templates-service/adaptivecards-templating-service/src/authproviders/AzureADProvider.ts

```diff
@@ -44,8 +44,12 @@ export class AzureADProvider implements AuthenticationProvider {
       }
     }
 
+    // Check expiry date on token
+    result = new Date() <= new Date(decodedToken.payload.exp * 1000); 
+
     // Check aud of token matches the client ID of env app
-    result = "#{CLIENT_ID_TOKEN}#" === decodedToken.payload.aud;
+    let appId = "#{CLIENT_ID_TOKEN}#";
+    result = appId === decodedToken.payload.appid || appId === decodedToken.payload.aud;
```

Comment:
After expiry date was checked on the token, the result of that check is not used. Modify the logic so if expiry date check failed, function will return false.